### PR TITLE
[OPAL-930] create repodata.json for every subdir

### PR DESCRIPTION
**Pull request by RR-USAF**
correctly create repodata.json for every subir in the solution.

### CODE REVIEW

**Review by JR-USAF — APPROVED** [review]
I didn't see anything that stuck out to me and all tests passed.

image

File: conda_vendor/conda_vendor.py [hunk 1]
```diff
@@ -70,45 +75,33 @@
...
-#  in a temporary directory).  After we solve, we remove all virtual_packages
-#  from the solution set.
-#
-#  the way we write repodata.json for each subdirectory will break if we
-#  vendor from multiple sources
-
 
-def _blue(s, bold=True):
-    click.echo(click.style(s, fg="blue", bg="black", bold=bold))
+def _blue(msg: str, bold: bool = True):
```
Comment (IM-USAF) [inline conda_vendor/conda_vendor.py:79]: Curious why you didn't break out some of this house-keeping stuff?

Comment (RR-USAF) [reply conda_vendor/conda_vendor.py:79]: i'm not sure what you mean, what have you got in mind?  I have no strong opinions about this.

File: conda_vendor/conda_vendor.py [hunk 2]
```diff
@@ -123,19 +116,31 @@ def _get_environment_name(environment_file: Path) -> str:
 
 
 def _get_virtual_packages(
```
Comment (IM-USAF) [inline conda_vendor/conda_vendor.py:118]: conda_vendor.py is pretty big. May help other contributors if you separate out some stuff, such as virtual package-related funcs.

File: conda_vendor/conda_vendor.py [hunk 3]
```diff
@@ -470,9 +506,9 @@ def yaml_dump_ironbank_manifest(package_list: List[FetchAction]):
 
         resources["resources"].append(resource)
 
-    yaml = YAML()
+    yaml_ = YAML()
```
Comment (IM-USAF) [inline conda_vendor/conda_vendor.py:509]: Is there some semantic reason for the trailing underscore?

Comment (RR-USAF) [reply conda_vendor/conda_vendor.py:509]: to not shadow import yaml at the top.  it was to silence a pylint warning.

File: conda_vendor/conda_vendor.py (hunk 2 above)
Comment (RR-USAF) [reply conda_vendor/conda_vendor.py:118]: i want to refactor more completely in the future, to separate the library stuff from the ui stuff. 

the counter argument is that it's ~800 lines with comments, ~600 lines of code (with at least 200 lines just doing ui), all in all, not too big. 

i'm not sure how much easier it is to understand if we just hide stuff away, and all the code kind of depends on the other code there, i couldn't find good module separation, yet.  

right now, all of conda vendor sits in conda_vendor.py which means there are no secrets.

File: conda_vendor/conda_vendor.py (hunk 1 above)
Comment (IM-USAF) [reply conda_vendor/conda_vendor.py:79]: Maybe put it in a `_terminal.py` or `_utils.py` or something.

File: conda_vendor/conda_vendor.py (hunk 2 above)
Comment (IM-USAF) [reply conda_vendor/conda_vendor.py:118]: I think refactoring can wait.

File: conda_vendor/conda_vendor.py (hunk 3 above)
Comment (IM-USAF) [reply conda_vendor/conda_vendor.py:509]: makes sense

**Review by IM-USAF — APPROVED** [review]

**Pull request merged**